CONTRIBUTING.md

```diff
@@ -22,11 +22,14 @@ Every code contribution to this project is highly valued and appreciated. I enco
    - PcapPlusPlus compiles successfully on your machine, including all unit-tests and examples
    - Unit-tests pass successfully on your machine (both `Tests/Packet++Test` and `Tests/Pcap++Test`)
    - All new APIs are well documented using Doxygen (please use @ for keywords)
-   - Make sure that pre-commit hooks are passing by using `pre-commit run --all-files`. For pre-commit hooks you need to install `cppcheck` on your system. Recommended and tested version is 2.7. You can install by using:
-     - For Linux (apt) `sudo apt install cppcheck`
-     - For Windows (using `choco`) `choco install cppcheck --version=2.7`
-     - For Windows (MSI install): https://github.com/danmar/cppcheck/releases/download/2.7/cppcheck-2.7-x64-Setup.msi
-     - Build from source: https://github.com/danmar/cppcheck/releases/tag/2.7
+   - Make sure that pre-commit hooks are passing by using `pre-commit run --all-files`. For pre-commit hooks you need to install `cppcheck`, `clang-format`, and `ruff` on your system. You can install them using the following commands:
+      - `pre-commit`: `pip install pre-commit`
+      - `cppcheck` (version 2.7 is recommended):
+         - For Linux (apt) `sudo apt install cppcheck`
+         - For Windows (using `choco`) `choco install cppcheck --version=2.7`
+         - For Windows (MSI install): https://github.com/danmar/cppcheck/releases/download/2.7/cppcheck-2.7-x64-Setup.msi
+         - Build from source: https://github.com/danmar/cppcheck/releases/tag/2.7
+      - `clang-format` and `ruff`: `pip install clang-format==18.1.6 ruff==0.6.5`
```

Comment:
Let's wait until we merge [this PR](https://github.com/seladb/PcapPlusPlus/pull/1584) to see if we need it

Reply:
@seladb pre-commit will run ruff in a venv, but I think users will also run ruff without using pre-commit.

Reply:
But we ask users to install `pre-commit` that includes `ruff`, do we really need this redundancy?

Reply:
Sure, removed ruff here.